src/client/Client.cc

```diff
@@ -10915,14 +10917,35 @@ int Client::_read_async(Fh *f, uint64_t off, uint64_t len, bufferlist *bl,
 
   const auto& conf = cct->_conf;
   Inode *in = f->inode.get();
+  std::unique_ptr<Context> io_finish = nullptr;
+  C_SaferCond *io_finish_cond = nullptr;
 
   ldout(cct, 10) << __func__ << " " << *in << " " << off << "~" << len << dendl;
 
+  if (onfinish != nullptr) {
+    io_finish.reset(new C_Read_Async_Finisher(this, onfinish, f, in,
+                                              f->pos, off, len));
+  }
+
   // trim read based on file size?
-  if (off >= in->size)
-    return 0;
-  if (len == 0)
+  if ((off >= in->size) || (len == 0)) {
+    if (onfinish != nullptr) 
```

Comment:
Did you actually mean ``if (onfinish == nulptr) { return 0; }``? At the moment, ``crf->complete(0)`` accesses a null pointer and crashes.